### ISSUE TITLE
Feature request: have optional outputs

### ISSUE THREAD

**Issue by giulia-berto** [brainlife/brainlife#87]
I don't know if it's possible, but what do you think of having optional outputs for Apps as we do have optional inputs?

**Comment by soichih** [comment]
Are you thinking something like.. maybe based on the configuration option, some output will not be generated? I think that's probably possible to do, but easiest thing to do is just register 2 different App and hard code such option for each versions.

More difficult scenario is when the App doesn't know if the output will be there or not until we execute it. If we don't know if output is going to be there or not until we run the App, then we won't be able to construct the workflow until the job finishes so that might not work.

**Comment by giulia-berto** [comment]
Yes, sometimes, depending on the configuration of the inputs, some outputs may be generated or not. When the number of the possible combinations inputs-output is "higher" (i.e. even 4 or 5), it will be uncomfortable to register 4 or 5 different Apps that do the same thing. However I understand that this would affect pipeline and reproducibility. It was just a thought, no need to have it right now if it's too complicated.
